modules/revcontentBidAdapter.js

```diff
@@ -0,0 +1,290 @@
+// jshint esversion: 6, es3: false, node: true
+'use strict';
+
+import {registerBidder} from '../src/adapters/bidderFactory';
+import * as utils from '../src/utils';
+import {ajax} from '../src/ajax';
+
+const BIDDER_CODE = 'revcontent';
+const NATIVE_PARAMS = {
+  title: {
+    id: 0,
+    name: 'title'
+  },
+  image: {
+    id: 3,
+    type: 3,
+    name: 'img'
+  },
+  sponsoredBy: {
+    id: 5,
+    name: 'data',
+    type: 1
+  }
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: ['banner', 'native', 'video'],
+  isBidRequestValid: function (bid) {
+    return (typeof bid.params.apiKey !== 'undefined' && typeof bid.params.userId !== 'undefined' && bid.hasOwnProperty('nativeParams'));
+  },
+  buildRequests: (validBidRequests, bidderRequest) => {
+    const userId = validBidRequests[0].params.userId;
+    const widgetId = validBidRequests[0].params.widgetId;
+    const apiKey = validBidRequests[0].params.apiKey;
+    var domain = validBidRequests[0].params.domain;
+    var host = validBidRequests[0].params.endpoint;
+
+    if (typeof host === 'undefined') {
+      host = 'trends.revcontent.com';
+    }
+
+    let serverRequests = [];
+    var refererInfo;
+    if (bidderRequest && bidderRequest.refererInfo) {
+      refererInfo = bidderRequest.refererInfo.referer;
+    }
+
+    if (typeof domain === 'undefined') {
+      domain = extractHostname(refererInfo);
```

Comment:
it `may` be possible that `refererInfo` is `undefined` at this point.

Which I believe could throw `Cannot read property 'indexOf' of undefined` inside the function `extractHostname`

May be a good idea to default this or skip this if `refererInfo` is not defined here.

Reply:
I am going to update the actual function to return an empty string in such event.